src/binary/ibc-setup/commands/init.ts

```diff
@@ -56,26 +62,55 @@ export async function run(options: Options) {
 
   const registryFilePath = path.join(options.home, registryFile);
   if (!fs.existsSync(registryFilePath)) {
-    try {
-      const registryFromRemote = await axios.get(
-        'https://raw.githubusercontent.com/confio/ts-relayer/main/demo/registry.yaml'
+    if (options.registryFrom) {
+      const existingRegistryFilePath = path.join(
+        options.registryFrom,
+        registryFile
       );
-      fs.writeFileSync(registryFilePath, registryFromRemote.data);
-    } catch (error) {
-      throw new Error(`Cannot fetch ${registryFile} from remote. ${error}`);
+      try {
+        fs.copyFileSync(existingRegistryFilePath, registryFilePath);
+        console.log(
+          `Copied existing registry from ${existingRegistryFilePath} to ${registryFilePath}.`
+        );
+      } catch (error) {
+        if (error.code === 'ENOENT') {
+          throw new Error(
+            `No such file: ${existingRegistryFilePath}. Make sure that "--registry-from" points at existing relayer's home dir.`
+          );
+        } else {
+          throw error;
+        }
+      }
+    } else {
+      try {
+        const registryFromRemote = await axios.get(
+          'https://raw.githubusercontent.com/confio/ts-relayer/main/demo/registry.yaml'
+        );
+        fs.writeFileSync(registryFilePath, registryFromRemote.data);
+        console.log(`Pulled default ${registryFile} from remote.`);
+      } catch (error) {
+        throw new Error(`Cannot fetch ${registryFile} from remote. ${error}`);
+      }
     }
   } else if (!fs.lstatSync(registryFilePath).isFile()) {
     throw new Error(`${registryFilePath} must be a file.`);
   }
 
+  if (!options.src || !options.dest) {
```

Comment:
Don't you want this check right at the beginning?

Reply:
No. The point is if `src` and `dest` are not specified, the home directory (with registry) should be created. See the message:
> `Exited early. Registry file downloaded to ${registryFilePath}. Please edit that file and add any chains you wish.
Then complete the initialization by running ibc-setup init --src <chain-1> --dest <chain-2>.`